modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java

```diff
@@ -214,50 +195,58 @@ public synchronized CompletableFuture<?> onTableCreated(
             TableImpl table,
             long causalityToken
     ) {
-        schemasVv.update(
-                causalityToken,
-                (schemas, e) -> {
-                    if (e != null) {
-                        return failedFuture(e);
-                    }
+        if (!busyLock.enterBusy()) {
+            return failedFuture(new IgniteInternalException(NODE_STOPPING_ERR, new NodeStoppingException()));
+        }
+        try {
+            schemasVv.update(
+                    causalityToken,
+                    (schemas, e) -> {
+                        if (e != null) {
+                            return failedFuture(e);
+                        }
 
-                    Map<String, IgniteSchema> res = new HashMap<>(schemas);
+                        Map<String, IgniteSchema> res = new HashMap<>(schemas);
 
-                    IgniteSchema schema = res.computeIfAbsent(schemaName, IgniteSchema::new);
+                        IgniteSchema schema = res.computeIfAbsent(schemaName, IgniteSchema::new);
 
-                    CompletableFuture<IgniteTableImpl> igniteTableFuture = convert(causalityToken, table);
+                        CompletableFuture<IgniteTableImpl> igniteTableFuture = convert(causalityToken, table);
 
-                    return tablesVv
-                            .update(
-                                    causalityToken,
-                                    (tables, ex) -> {
-                                        if (ex != null) {
-                                            return failedFuture(ex);
+                        return tablesVv
+                                .update(
+                                        causalityToken,
+                                        (tables, ex) -> {
+                                            if (ex != null) {
+                                                return failedFuture(ex);
+                                            }
+
+                                            Map<UUID, IgniteTable> resTbls = new HashMap<>(tables);
+
+                                            return igniteTableFuture
+                                                    .thenApply(igniteTable -> {
+                                                        resTbls.put(igniteTable.id(), igniteTable);
+
+                                                        return resTbls;
+                                                    });
```

Comment:
this will be asynchronous part in this update, possibly needs to be wrapped with busy lock